core/services/gateway/handlers/vault/handler.go

```diff
@@ -271,15 +288,50 @@ func (h *handler) handleSecretsCreate(ctx context.Context, ar activeRequest) err
 		err := h.don.SendToNode(ctx, node.Address, &ar.req)
 		if err != nil {
 			nodeErrors = append(nodeErrors, err)
-			h.lggr.Errorw("error sending request to node", "node", node.Address, "error", err)
+			l.Errorw("error sending request to node", "node", node.Address, "error", err)
+		}
+	}
+
+	if len(nodeErrors) == len(h.donConfig.Members) && len(nodeErrors) > 0 {
+		return h.sendResponse(ctx, ar, h.errorResponse(ar.req, api.FatalError, errors.New("failed to forward user request to nodes")))
+	}
+
+	l.Debugw("successfully forwarded request to Vault nodes")
+	return nil
+}
+
+func (h *handler) handleSecretsUpdate(ctx context.Context, ar activeRequest) error {
+	l := logger.With(h.lggr, "method", ar.req.Method, "requestId", ar.req.ID)
+
+	req := &vault.UpdateSecretsRequest{}
```

Comment:
Need some fields validation here.
See how we validate fields for Create request.
Also, this is coming from user. Can the user provide directly in this format? We might need to define user's schema in gateway/handlers/vault/api.go

Reply:
I thought about adding validation here, but in the end decided against it since we already do validation on the node side and I'd just end up duplicating the same checks.

Do you still see value in adding it here too? If so I can add

Reply:
Ok to skip it.
I just want to make all methods consistent with each other.

Thinking more, it might be better to do as much error validation earlier as possible, since the Vault DON consensus is costly, and prone to DDoS attacks.

Don't have strong opinion yet, but what do you think?